code/game/objects/structures/ladders.dm

```diff
@@ -123,39 +177,77 @@
 		return
 	use(user)
 
+/obj/structure/ladder/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	use(user, going_up = FALSE)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+//Not be called when right clicking as a monkey. attack_hand_secondary() handles that.
 /obj/structure/ladder/attack_paw(mob/user, list/modifiers)
-	return use(user)
+	use(user)
+	return TRUE
 
 /obj/structure/ladder/attack_alien(mob/user, list/modifiers)
-	return use(user)
+	var/is_right_clicking = LAZYACCESS(modifiers, RIGHT_CLICK)
+	use(user, going_up = !is_right_clicking)
+	return TRUE
+
+/obj/structure/ladder/attack_larva(mob/user, list/modifiers)
+	var/is_right_clicking = LAZYACCESS(modifiers, RIGHT_CLICK)
+	use(user, going_up = !is_right_clicking)
+	return TRUE
+
+/obj/structure/ladder/attack_animal(mob/user, list/modifiers)
+	var/is_right_clicking = LAZYACCESS(modifiers, RIGHT_CLICK)
+	use(user, going_up = !is_right_clicking)
+	return TRUE
 
-/obj/structure/ladder/attack_larva(mob/user)
-	return use(user)
+/obj/structure/ladder/attack_slime(mob/user, list/modifiers)
```

Comment:
Can you replace these with some more core `attack_secondary`? I want to get rid of all the RIGHT_CLICK checks.